components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -354,4 +354,29 @@ private URI getResourceLocationURI(CreateResourceDTO response) throws URISyntaxE
         return new URI(RESOURCE_PATH + "/" + response.getResourceId());
 
     }
+
+    private boolean isResourceOwner(String resourceId, MessageContext context) {
+
+        String consumerKey = getConsumerKey(context);
+        String userNameWithDomain = getUserNameWithDomain(context);
+        String userDomain = null;
+        String resourceOwner = null;
+        if (userNameWithDomain != null) {
+            resourceOwner = UserCoreUtil.removeDomainFromName(userNameWithDomain);
+            userDomain = IdentityUtil.extractDomainFromName(userNameWithDomain);
+        }
+
+        try {
+            if (resourceOwner != null && consumerKey != null) {
+                List<String> resourceList = ResourceUtils.getResourceService()
+                        .getResourceIds(resourceOwner, userDomain, consumerKey);
+                if (resourceList.contains(resourceId)) {
+                    return true;
+                }
+            }
+        } catch (UMAException e) {
+            handleErrorResponse("Error when retrieving resource for client: " + consumerKey, e);
```

Comment:
is this line meaning full ?

[Error] Error when retrieving resource for client : null
or
[Error] Error when retrieving resource for client : 34820hnbfsamf023u41qwe

Reply:
PR updated with the fix.